lib/puppet/type/image_streamer_deployment_group.rb

```diff
@@ -0,0 +1,40 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+Puppet::Type.newtype(:image_streamer_deployment_group) do
+  desc "Image Streamer's Deployment Group"
+
+  # :nocov:
+  ensurable do
+    defaultvalues
```

Comment:
Maybe removing defaultvalues from here is better in this resource than actually defining create/delete as just a raise to mention that they are not supported.